packages/smooth_app/lib/cards/product_cards/knowledge_panels/knowledge_panel_summary_card.dart

```diff
@@ -4,22 +4,29 @@ import 'package:smooth_app/cards/data_cards/score_card.dart';
 import 'package:smooth_app/cards/product_cards/knowledge_panels/knowledge_panel_title_card.dart';
 import 'package:smooth_app/helpers/score_card_helper.dart';
 
+import 'package:smooth_ui_library/util/ui_helpers.dart';
+
 class KnowledgePanelSummaryCard extends StatelessWidget {
   const KnowledgePanelSummaryCard(this.knowledgePanel);
 
   final KnowledgePanel knowledgePanel;
 
   @override
   Widget build(BuildContext context) {
-    if (knowledgePanel.type == KnowledgePanelType.SCORE) {
+    if (knowledgePanel.titleElement == null) {
+      return EMPTY_WIDGET;
+    }
+    if (knowledgePanel.titleElement!.type == TitleElementType.GRADE) {
       return ScoreCard(
-        iconUrl: knowledgePanel.titleElement.iconUrl!,
-        description: knowledgePanel.titleElement.title,
-        cardEvaluation: getCardEvaluationFromKnowledgePanel(knowledgePanel),
+        iconUrl: knowledgePanel.titleElement!.iconUrl!,
+        description: knowledgePanel.titleElement!.title,
+        cardEvaluation: getCardEvaluationFromKnowledgePanelTitleElement(
+          knowledgePanel.titleElement!,
+        ),
       );
     }
     return KnowledgePanelTitleCard(
-      knowledgePanelTitleElement: knowledgePanel.titleElement,
+      knowledgePanelTitleElement: knowledgePanel.titleElement!,
       evaluation: knowledgePanel.evaluation,
     );
```

Comment:
A `switch` syntax would be cleaner, as you're using an `enum` as input.

Reply:
Done